x/payment/keeper/stream_record.go

```diff
@@ -158,7 +164,12 @@ func (k Keeper) UpdateStreamRecord(ctx sdk.Context, streamRecord *types.StreamRe
 		streamRecord.LockBalance = streamRecord.LockBalance.Add(change.LockBalanceChange)
 		streamRecord.StaticBalance = streamRecord.StaticBalance.Sub(change.LockBalanceChange)
 		if streamRecord.LockBalance.IsNegative() {
-			return fmt.Errorf("lock balance can not become negative, current: %s", streamRecord.LockBalance)
+			if ctx.IsUpgraded(upgradetypes.Pawnee) {
```

Comment:
so is here.

Reply:
Yes. It could be negative. As when we locking more fund into buffer balance, if the static balance is negative. The new needs to pay for it.

<img width="1140" alt="image" src="https://github.com/bnb-chain/greenfield/assets/61674316/b8fa6fa0-5420-4fea-a086-296d57b7de20">